### PR TITLE
externals: Update dynarmic to e814b4

### LINKED ISSUE #4766: Compilation error with GCC 9.1 in fmt
[citra fail build .log](https://github.com/citra-emu/citra/files/3154060/citra.fail.build.log)

### PR DESCRIPTION
**Pull request by MerryMage**
Closes #4766.

---
This change is [Reviewable](https://reviewable.io/reviews/citra-emu/citra/4850)

### CODE REVIEW

**Comment by wwylele** [comment]
Looks like some indirect include is exposed by this

**Comment by Dragios** [comment]
Performance regression with this PR.
Speed from 90% (**Canary-1410**) drops to 70% (**Canary-1411**) in **Pokémon X** intro FMV with *Intel® Core™ i5-2500 Processor*

**Comment by B3n30** [comment]
removed from canary due to multiple users reporting speed regressions

**Comment by MerryMage** [comment]
Performance regression hopefully fixed by [dc8e6c...e814b4](https://github.com/MerryMage/dynarmic/compare/dc8e6ccf7459c592fcab309b27fef1ea25000e22...e814b47812723b384c9d1fbeeb72d6d9a28d0d8e).

**Comment by wwylele** [comment]
regression #4857 might be caused by this. (Nightly fine, canary 1413 fine, canary 1414 hang, the delta is this PR)

**Comment by Developer-exe** [comment]
The issue causes to freeze the following games:
- Project X Zone
- Project X Zone 2: Brave New World

After one of these games freeze, the log file shows this warning repetitively:
`Service.Y2R <Warning> core/hle/service/y2r_u.cpp:SetTransferEndInterrupt:198: (STUBBED) called`

The background music continues to run, but nothing seems to be happening on the screen, pushing buttons won't trigger any visual animation/sound effects. The GPU reports the game running smoothly at 60 FPS, although the image isn't refreshing at all.

_Project X Zone (EUR)_: [citra_log.txt](https://github.com/citra-emu/citra/files/3434666/citra_log.txt)
_Project X Zone 2 (EUR_): [citra_log.txt](https://github.com/citra-emu/citra/files/3433030/citra_log.txt)

**Comment by MerryMage** [comment]
Pused an instruction bugfix (MerryMage/dynarmic@e656eb0b8213108f4f7568f770bbe080cd6144fe).

**Comment by Dragios** [comment]
The previous commit has a little better FPS in FMV though. I guess 1 - 2 FPS loss is not really that big deal for freeze bug fix.

**Comment by Developer-exe** [comment]
Updated to _Canary 1418_ (_8474eee_) to try the bugfix, the issue persists ([citra_log.txt](https://github.com/citra-emu/citra/files/3438973/citra_log.txt))
`Service.Y2R <Warning> core/hle/service/y2r_u.cpp:SetTransferEndInterrupt:198: (STUBBED) called`
Video showing the issue: [https://youtu.be/DsLOkOFFl-M](https://youtu.be/DsLOkOFFl-M)

**Comment by Developer-exe** [comment]
I also noticed one thing while testing the game on _Nightly 1315_ (_45ff10c_), where the permanent freeze doesn't occur. When a minor freeze happens (the game slows down to 30-60% for around 1 second), the log file shows the exact same warning as in Canary, but less number of times (18x in the log I attached below) and then continues the emulation without any issue. It looks like this:
```
[  12.726795] Service.Y2R <Warning> core/hle/service/y2r_u.cpp:SetTransferEndInterrupt:198: (STUBBED) called
[  14.934428] Service.AC <Warning> core/hle/service/ac/ac.cpp:SetClientVersion:166: (STUBBED) called, version: 0x0B0101C8
[  14.934522] Service.AM <Warning> core/hle/service/am/am.cpp:ListDataTitleTicketInfos:900: (STUBBED) ticket_count=0x00000001, title_id=0x0004008c00162000, start_index=0x00000000
[  14.934743] Service.AC <Warning> core/hle/service/ac/ac.cpp:IsConnected:157: (STUBBED) called unk=0x00000000 descriptor=0x00000020 param=0x0000000B
[  18.347958] Render.OpenGL <Warning> video_core/renderer_opengl/renderer_opengl.cpp:DebugHandler:535: API PERFORMANCE 131218: Program/shader state performance warning: Vertex shader in program 30 is being recompiled based on GL state.
```

Full log file of _Nightly 1315_ (_45ff10c_): [citra_log.txt](https://github.com/citra-emu/citra/files/3439042/citra_log.txt)

**Comment by MerryMage** [comment]
MerryMage/dynarmic@3aecba983327c1b20d6e392b6e0edb8d36f3607b now resets exclusive state on context change to prevent thread deadlocking weirdness.

**Comment by MerryMage** [comment]
@Developer-exe Unable to reproduce anymore on the current canary, could you confirm?

**Comment by Developer-exe** [comment]
Testing _Canary 1425_ (_71ef668_), works great so far. The log file still shows the issue, but it doesn't affect the gameplay. I'd compare that to the state of _Nightly 1315_ (_45ff10c_) described above, but the warning is showing a lot (approx. 30 times a second), but the game runs without issues so far.

The state of _Nightly 1315_ (_45ff10c_) mentioned in this reply:
> When a minor freeze happens (the game slows down to 30-60% for around 1 second), the log file shows the exact same warning as in Canary
The log file is very bloated by this, but the game runs fine.

Here's the log (just in case if anyone wonders about the warning showing too many times): [citra_log.txt](https://github.com/citra-emu/citra/files/3442745/citra_log.txt)

I'll continue playing the game to make sure the hang issue is really gone.
Many thanks on working on the fix.

**Comment by B3n30** [comment]
Just FYI: those entries in the log are totally unrelated to the issue... They just appear more often because they run in another thread. So if the one thread hangs this thread spams those log entries. But they are no sign of anything wrong.

**Comment by Developer-exe** [comment]
> 
> 
> Just FYI: those entries in the log are totally unrelated to the issue... They just appear more often because they run in another thread. So if the one thread hangs this thread spams those log entries. But they are no sign of anything wrong.

Okay, I didn't know that, thanks for telling me. I thought it might cause further problems in the future, so that's why I wanted to let you know.

**Comment by Developer-exe** [comment]
> 
> 
> @Developer-exe Unable to reproduce anymore on the current canary, could you confirm?

@MerryMage Confirmed. I can no longer reproduce the issue.

- _Project X Zone_ (_EUR_, **15 minutes**) - **can't reproduce the issue**
- _Project X Zone 2_ (_EUR_, **1 hour+**) - **can't reproduce the issue**
- _Project X Zone 2 OGSE_ (_JP_, **15 minutes**) - **can't reproduce the issue**

Looks good to go. Thanks a lot for your fix!

File: .travis/macos/build.sh [hunk 1]
```diff
@@ -2,13 +2,11 @@
...
 set -o pipefail
 
-export MACOSX_DEPLOYMENT_TARGET=10.13
+export MACOSX_DEPLOYMENT_TARGET=10.14
 export Qt5_DIR=$(brew --prefix)/opt/qt5
 export PATH="/usr/local/opt/ccache/libexec:$PATH"
 
 # TODO: Build using ninja instead of make
 mkdir build && cd build
-cmake .. -DCMAKE_OSX_ARCHITECTURES="x86_64;x86_64h" -DCMAKE_BUILD_TYPE=Release -DENABLE_QT_TRANSLATION=ON -DCITRA_ENABLE_COMPATIBILITY_REPORTING=${ENABLE_COMPATIBILITY_REPORTING:-"OFF"} -DENABLE_COMPATIBILITY_LIST_DOWNLOAD=ON -DUSE_DISCORD_PRESENCE=ON -DENABLE_FFMPEG=ON
```
Comment (wwylele) [inline .travis/macos/build.sh]: What does this change mean and is it intentional? Could this break citra on some machines?

File: .travis/macos/build.sh [hunk 2]
```diff
@@ -2,13 +2,11 @@
...
+cmake .. -DCMAKE_OSX_ARCHITECTURES="x86_64h" -DCMAKE_BUILD_TYPE=Release -DENABLE_QT_TRANSLATION=ON -DCITRA_ENABLE_COMPATIBILITY_REPORTING=${ENABLE_COMPATIBILITY_REPORTING:-"OFF"} -DENABLE_COMPATIBILITY_LIST_DOWNLOAD=ON -DUSE_DISCORD_PRESENCE=ON -DENABLE_FFMPEG=ON
 make -j4
-
-ctest -VV -C Release
```
Comment (wwylele) [inline .travis/macos/build.sh]: Is removing test because of the timeout?

File: .travis/macos/build.sh [hunk 3]
```diff
@@ -2,13 +2,11 @@
 
 set -o pipefail
 
-export MACOSX_DEPLOYMENT_TARGET=10.13
+export MACOSX_DEPLOYMENT_TARGET=10.14
```
Comment (B3n30) [inline .travis/macos/build.sh]: Is this change necessary? This will reduce the compatibility of Mac's heavily

Comment (MerryMage) [reply .travis/macos/build.sh]: PR requiring this is https://github.com/MerryMage/dynarmic/pull/497.

I will have to check what the interface of std::variant is on 10.12/10.13 to see if this is possible.

File: .travis/macos/build.sh (hunk 1 above)
Comment (MerryMage) [reply .travis/macos/build.sh]: This build only works on Macs about 2013 and later. Done to reduce build times.

File: .travis/macos/build.sh (hunk 2 above)
Comment (MerryMage) [reply .travis/macos/build.sh]: CI is not a post-Haswell machine.

File: .travis/macos/build.sh (hunk 3 above)
Comment (B3n30) [reply .travis/macos/build.sh]: if it is about std::variant. on osx below 10.14 you can't use any part that throws a bad_variant_access exception. this means basically `std::get` isn't available. The solution here is to use `std::get_if` which in return also avoids handling with catching the exception

Comment (jroweboy) [reply .travis/macos/build.sh]: Looks like merry has already updated it https://github.com/MerryMage/dynarmic/commit/087a74417abfb0a8ae3bc1463d0d476a9bf94e53 and just needs to update this PR with the new commits

**Comment by zhaowenlan1779** [comment]
Temorarily untagging due to conflicts

**Comment by MerryMage** [comment]
Closing until future update.
